src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -44,6 +48,26 @@
    (every? keyword? (keys service-graph))
    (every? (some-fn ifn? service-graph?) (vals service-graph))))
 
+(defn inc-restart-counter!
+  "Increments the counter in a restart-file for purposes of supporting HUP behavior"
+  [restart-file-path]
+  (when (some? restart-file-path)
```

Comment:
It looks like restart-file is not a collection, but is going to be a string? I think if you make the change I suggested in start to use `when-let` instead of let, you can remove this `when` call.